packages/manager/modules/hub/src/components/carousel/carousel.controller.js

```diff
@@ -0,0 +1,55 @@
+import first from 'lodash/first';
+import filter from 'lodash/filter';
+import findIndex from 'lodash/findIndex';
+import map from 'lodash/map';
+import set from 'lodash/set';
+
+import { Environment } from '@ovh-ux/manager-config';
+
+export default class {
+  /* @ngInject */
+  constructor($window, OvhApiNotificationAapi, TranslateService) {
+    this.$window = $window;
+    this.OvhApiNotificationAapi = OvhApiNotificationAapi;
+    this.TranslateService = TranslateService;
+    this.REGION = Environment.getRegion();
+  }
+
+  $onInit() {
+    return this.OvhApiNotificationAapi.query(
+      this.TranslateService.getUserLocale(),
+      this.REGION,
+    )
+      .$promise.then((notificationList) => {
+        this.notifications = map(
+          filter(
+            notificationList,
+            (notification) =>
+              notification.level === 'warning' ||
+              notification.level === 'error',
+          ),
+          (notification) => ({
+            ...notification,
+            active: false,
+          }),
+        );
+        set(first(this.notifications), 'active', true);
+      })
+      .catch(() => undefined);
+  }
+
+  nextAlert() {
+    this.activeIndex = findIndex(this.notifications, { active: true });
+    set(this.notifications[this.activeIndex], 'active', false);
```

Comment:
notifications array will be passed to the component. don't change the objects in the array. You can use active index to track the active notification or a variable like activeNotification which points to the active notification.

Reply:
I dont see any other place in hub page which uses notifications. Only navbar has it and the data is not shared with it. Since the only use is for this component i thought its better if we pull data inside the component itself ?